spark/src/test/java/org/apache/iceberg/spark/source/TestSparkReaderDeletes.java

```diff
@@ -159,4 +168,72 @@ public void testEqualityDeleteWithFilter() throws IOException {
 
     Assert.assertEquals("Table should contain no rows", 0, actual.size());
   }
+
+  @Test
+  public void testReadDeleteRow() throws IOException {
```

Comment:
Another point, I think we'd better to move those newly added unit tests to the abstracted `DeleteReadTests` as possible as we can.  Then we don't have to introduce this test again for other engines.

Reply:
Make sense to me. I plan to add an abstract function to read delete rows and that should be implemented in the engine specific test classes. While the engine specific read logic is not ready, I think we can refactor when those are ready.